sdk/core/core-https/src/pipeline.ts

```diff
@@ -343,3 +365,100 @@ class HttpsPipeline implements Pipeline {
 export function createEmptyPipeline(): Pipeline {
   return HttpsPipeline.create();
 }
+
+/**
+ * Options that allow configuring redirect behavior.
+ */
+export interface PipelineRedirectOptions extends RedirectPolicyOptions {
+  /**
+   * If true, disables automatic following of redirects.
+   */
+  disable?: boolean;
+}
+
+/**
+ * Defines options that are used to configure the HTTP pipeline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * The HttpsClient implementation to use for outgoing HTTP requests.
+   * Defaults to DefaultHttpsClient.
+   */
+  httpClient?: HttpsClient;
+
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: ExponentialRetryPolicyOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxySettings;
+
+  /**
+   * Options for how HTTP connections should be maintained for future
+   * requests.
+   */
+  keepAliveOptions?: KeepAlivePolicyOptions;
+
+  /**
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: PipelineRedirectOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+
+  /**
+   * Configure whether to decompress response according to Accept-Encoding header (node-fetch only)
+   */
+  decompressResponse?: boolean;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options Options to configure a custom pipeline.
+ */
+export function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline {
+  const pipeline = HttpsPipeline.create();
+
+  if (isNode) {
+    pipeline.addPolicy(proxyPolicy(options.proxyOptions));
+
+    if (options.decompressResponse === false) {
+      pipeline.addPolicy(disableResponseDecompressionPolicy());
+    }
+  }
+
+  pipeline.addPolicy(formDataPolicy());
+  pipeline.addPolicy(tracingPolicy(options.userAgentOptions));
+  pipeline.addPolicy(keepAlivePolicy(options.keepAliveOptions));
+  pipeline.addPolicy(userAgentPolicy(options.userAgentOptions));
+  pipeline.addPolicy(setClientRequestIdPolicy());
+  pipeline.addPolicy(throttlingRetryPolicy(), { phase: "Retry" });
+  pipeline.addPolicy(systemErrorRetryPolicy(options.retryOptions), { phase: "Retry" });
+  pipeline.addPolicy(exponentialRetryPolicy(options.retryOptions), { phase: "Retry" });
```

Comment:
Not directly related to this PR, but have you thought about potentially creating a single retry policy, or something on the Request object to keep track of how many retries we've used up? Noticing while looking at these policies that they all keep track of retryCounts separately, so you could theoretically end up with a lot more retries occurring than you (the user) configured.

Reply:
Right, something we don't think about today is how many times a request has been retried across different policies. We could track this per-request if we wanted. 

Also there is definitely opportunity for code-reuse between systemErrorRetryPolicy and exponentialRetryPolicy.

Reply:
Filed #9569 to revisit this decision so we don't get too bogged down in this review